synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -337,6 +378,10 @@ void Widget_Timetrack::set_action_state(Widget_Timetrack::ActionState action_sta
 {
 	waypoint_sd.set_action(action_state);
 }
+void Widget_Timetrack::set_interpolation(synfig::Interpolation type)
+{
+	waypoint_sd.set_widget_interp(type);
```

Comment:
Strange call here: you make Dock tool button call `Widget_Timetrack::set_interpolation()` that only calls `Widget_Timetrack::WaypointSD::set_widget_interp()` that, in its turn, only calls `Widget_Timetrack::interpolate_selected()`.

Why  doesn't `Widget_Timetrack::set_interpolation()` contents is directly the one of `Widget_Timetrack::interpolate_selected()`?

Reply:
hmm, I dont really know what I was thinking while doing that xD, anyways I removed the extra unnecessary functions and just left `Widget_Timetrack::interpolate_selected()`